.github/workflows/ci.yaml

```
@@ -0,0 +1,55 @@
+name: kubectl plugin test
+on:
+  pull_request:
+
+defaults:
+  run:
+    # reference: https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#using-a-specific-shell
+    shell: bash --noprofile --norc -eo pipefail -x {0}
+
+jobs:
+  Plugin-test:
+    runs-on: ubuntu-latest
+    steps:
+    - name: checkout
+      uses: actions/checkout@v2
+      with:
+        fetch-depth: 0
+
+    - name: setup minikube
+      uses: manusa/actions-setup-minikube@v2.4.2
+      with:
+        minikube version: 'v1.21.0'
+        kubernetes version: 'v1.19.2'
+        start args: --memory 6g --cpus=2
+        github token: ${{ secrets.GITHUB_TOKEN }}
+
+    - name: print k8s cluster status
+      run:  |
+        minikube status
+        kubectl get nodes
+
+    - name: use local disk
+      run: tests/github-action-helper.sh use_local_disk
+
+    - name: deploy rook cluster
+      run: tests/github-action-helper.sh deploy_rook
+
+    - name: wait for operator pod to be ready
+      run: tests/github-action-helper.sh wait_for_pod_to_be_ready_state
+
+    - name: Test pluging with ceph commands
+      run: |
+        # when we have the kubectl krew plugin available we can use like `kubectl rook-ceph` but for now in local deployment we have to use `kubectl rook` as kubectl plugin has this limitation.
+        # Doc link to clear the comment above:
+          # https://krew.sigs.k8s.io/docs/developer-guide/develop/naming-guide/
+          # https://kubernetes.io/docs/tasks/extend-kubectl/kubectl-plugins/#naming-a-plugin
+          # https://krew.sigs.k8s.io/docs/developer-guide/distributing-with-krew/
+        sudo mv kubectl-rook-ceph.sh /usr/local/bin/kubectl-rook
+        # run ceph commands with the plugin
+        kubectl rook ceph status
```

Based on [this discussion](https://github.com/rook/kubectl-rook-ceph/issues/2#issuecomment-921111474), I'm thinking we would want a `toolbox` command for anything that's a ceph command. We could still implement it to run in the operator pod, but I think `toolbox` is a good keyword for this.
```suggestion
        kubectl rook-ceph toolbox ceph status
```

---

Personally, I am in favor of having both `kubectl rook-ceph ceph <ceph cmd>` and `kubectl rook-ceph toolbox <any command>` commands. I think the `kubectl rook-ceph ceph` command is a useful shortcut, and we can always run it in the operator pod. However, I do think the `kubectl rook-ceph toolbox` command allows this case in addition to others (and therefore has more flexibility), and it might be a better command to start with.

IMO, I think we should continue with Subham's PR as-is for now so he doesn't have to rework it as much as would be required for the `toolbox` command. We can always remove the `ceph` command in the future if we come to a consensus that just the `toolbox` command is our preferred user experience. Right now I think having a working command is a good starting place.

---

Agreed we really should have this optimization instead of forcing the `toolbox` command. The ceph commands will really be a common use of this tool so it should be simple. To be clear, this means:
- `kubectl rook-ceph ceph <args>`: Any ceph command. If the toolbox is running it could be run there, otherwise fall back and run in the operator pod.
- `kubectl rook-ceph toolbox <command> <args>`: Any command that could be run in the toolbox. (for a future PR)

---

I think `kubectl rook-ceph toolbox <command> <args>:` this will make the command longer and also IMO I think we just want to run command and get output whether it ran in operator pod or toolbox pod.
But we can discuss this for future PRs as you said.